test/components/radio_test.rb

```diff
@@ -0,0 +1,208 @@
+        },
+        {
+          value: "govuk-verify",
+          text: "Use GOV.UK Verify",
+          bold: true
+        }
+      ]
+    )
+
+    assert_select ".app-c-radio__input[name=radio-group-bold-labels]"
+    assert_select ".app-c-radio .app-c-label--bold"
+  end
+
+  test "renders radio-group with hint text" do
+    render_component(
+      name: "radio-group-hint-text",
+      items: [
+        {
+          value: "government-gateway",
+          hint_text: "You'll have a user ID if you've signed up to do things like sign up Self Assessment tax return online.",
+          text: "Use Government Gateway"
+        },
+        {
+          value: "govuk-verify",
+          hint_text: "You'll have an account if you've already proved your identity with a certified company, such as the Post Office.",
+          text: "Use GOV.UK Verify"
+        }
+      ]
+    )
+
+    assert_select ".app-c-radio__input[name=radio-group-hint-text]"
+    assert_select ".app-c-radio:first-child .app-c-radio__label", text: "Use Government Gateway"
+    assert_select ".app-c-radio:first-child .app-c-label__hint", text: "You'll have a user ID if you've signed up to do things like sign up Self Assessment tax return online."
+    assert_select ".app-c-radio:last-child .app-c-radio__label", text: "Use GOV.UK Verify"
+    assert_select ".app-c-radio:last-child .app-c-label__hint", text: "You'll have an account if you've already proved your identity with a certified company, such as the Post Office."
+  end
+
+  test "renders radio-group with checked option" do
+    render_component(
+      name: "radio-group-checked-option",
+      items: [
+        {
+          value: "government-gateway",
+          text: "Use Government Gateway"
+        },
+        {
+          value: "govuk-verify",
+          text: "Use GOV.UK Verify",
+          checked: true
+        }
+      ]
+    )
+
+    assert_select ".app-c-radio__input[name=radio-group-checked-option]"
+    assert_select ".app-c-radio__input[checked]", value: "govuk-verify"
+  end
+
+  test "renders radio-group with custom id prefix" do
+    render_component(
+      id_prefix: 'custom',
+      name: "radio-group-custom-id-prefix",
+      items: [
+        {
+          value: "government-gateway",
+          text: "Use Government Gateway"
+        },
+        {
+          value: "govuk-verify",
+          text: "Use GOV.UK Verify"
+        }
+      ]
+    )
+
+    assert_select ".app-c-radio__input[name=radio-group-custom-id-prefix]"
+    assert_select ".app-c-radio__input[id=custom-0]", value: "government-gateway"
+    assert_select ".app-c-radio__label[for=custom-0]", text: "Use Government Gateway"
+    assert_select ".app-c-radio__input[id=custom-1]", value: "govuk-verify"
+    assert_select ".app-c-radio__label[for=custom-1]", text: "Use GOV.UK Verify"
+  end
+
+  # This component can be interacted with, so use integration tests for these cases.
+  class RadioIntegrationTest < ActionDispatch::IntegrationTest
```

Comment:
I think that these tests should be in a separate file under integration tests.

Reply:
I'd prefer to have tests colocated so they're easy to update and maintain, components don't really fit well with Rails conventions in this respect.

Ideally all component related files would be together, but that's for another time.